dotCMS/src/main/java/com/dotmarketing/common/reindex/ReindexThread.java

```diff
@@ -315,35 +360,44 @@ public static void unpause() {
         }
     }
 
-    private static void unpauseImpl() {
+    private static synchronized void unpauseImpl() {
+        ReindexThread instance = getInstance();
+        ThreadState currentState = instance.state.get();
 
-        ThreadState state = getInstance().state.get();
-        if (state == ThreadState.PAUSED) {
+        if (currentState == ThreadState.PAUSED) {
             Logger.info(ReindexThread.class, "--- Unpausing reindex thread ");
             cache.get().remove(REINDEX_THREAD_PAUSED);
-            getInstance().state(ThreadState.RUNNING);
-        } else if (state == ThreadState.STOPPED) {
+            instance.state.compareAndSet(ThreadState.PAUSED, ThreadState.RUNNING);
+        } else if (currentState == ThreadState.STOPPED) {
             Logger.info(ReindexThread.class, "--- Recreating ReindexThread from stopped");
             OSGISystem.getInstance().initializeFramework();
             Logger.infoEvery(ReindexThread.class, "--- ReindexThread Running", 60000);
             cache.get().remove(REINDEX_THREAD_PAUSED);
-            final Thread thread = new Thread(getInstance().ReindexThreadRunnable,
-                    "ReindexThreadRunnable");
 
             final DotSubmitter submitter = DotConcurrentFactory.getInstance()
-                    .getSubmitter("ReindexThreadSubmitter",
+                    .getSubmitter(
+                            "ReindexThreadSubmitter",
                             new DotConcurrentFactory.SubmitterConfigBuilder()
                                     .poolSize(1)
                                     .maxPoolSize(1)
                                     .queueCapacity(2)
-                                    .rejectedExecutionHandler(
-                                            new ThreadPoolExecutor.DiscardOldestPolicy())
-                                    .build()
-                    );
-            submitter.submit(thread);
-            getInstance().state(ThreadState.RUNNING);
-        }
+                                    .rejectedExecutionHandler(new ThreadPoolExecutor.DiscardOldestPolicy())
+                                    .build());
 
+            try {
+                submitter.submit(instance.reindexThreadRunnable);
+                // Set state to RUNNING only after successful submission
+                instance.state.compareAndSet(ThreadState.STOPPED, ThreadState.RUNNING);
+            } catch (RejectedExecutionException e) {
+                // Handle RejectedExecutionException specifically
+                Logger.error(ReindexThread.class, "Failed to submit reindex thread: RejectedExecutionException", e);
+                instance.state.set(ThreadState.STOPPED);
```

Comment:
This seems like new logic - are we sure we want to STOP here on exception?  Does the `DiscardOldestPolicy` even throw?

Reply:
Yes, this is the exception that gets thrown when an item cannot be added to the queue.   But I see the point and in the case where we already have a reindex in process and we have another queued up.  It also looks like our submitter already catches and rethrows the RejectedExecutionException so there is no point in separately handling the catch.    Unfortunately this makes it difficult to know whether an exception to submit was due to a capacity issue or some unrelated issue so currently we have to assume that the thread is working correctly ignore any failure here,  on the flip side if there is an execption thrown the indexer thread may have a temporary error and we will not retry assuming that we are still in a RUNNING state.